### ISSUE TITLE
Using the functional API for training only

### ISSUE THREAD

**Issue by epetrovski** [vecxoz/vecstack#34]
There doesn't seem to be a way to use the functional API just for training a model - since `X_test=` is a required argument. However, if I've already tested my 2nd level model, I think I should be able to train a model on the full data set.

To be clear, I would like to be able to just do the following:

```python
from vecstack import stacking

# Get your data

# Initialize 1st level estimators
models = [LinearRegression(),
          Ridge(random_state=0)]

# Get your stacked features in a single line
S_train = stacking(models, X_train, y_train, regression=True, verbose=2)

# Use 2nd level estimator with stacked features
```

Am I missing something?

**Comment by vecxoz** [comment]
Thanks! Your observation is correct. `X_test` should be optional. I will modify the code.

But the functionality you've mentioned is already available via `mode='oof'`.  
You can avoid passing `X_test` by explicitly setting it to `None`. (Somehow this fact was not documented.)  

```python
S_train, _ = stacking(models, X_train, y_train, None, mode='oof', regression=True)
```

Note that `stacking` function always returns tuple of 2 elements for consistency (in our case 2nd element will be `None`). Also note that `stacking` function does not save trained models. So if you eventually want to get `S_test` you have to rerun the function i.e. retrain all models. If you want to save trained models and apply them later Scikit-learn API would be better choice:  

```python
from vecstack import StackingTransformer
```

**Comment by epetrovski** [comment]
Super, thank you very much.

Your documentation is extensive by the way, which is greatly appreciated. It's even good for learning the details about stacking.

**Comment by vecxoz** [comment]
Eventually I decided to leave `X_test` as **required** argument. But I filled the gap in doc string of `stacking` function by adding notice saying `X_test` can be `None` for `mode='oof'`. This decision is dictated by the optimal set of default arguments serving the most common use case. In the most common use case user needs both `S_train` and `S_test` (default `mode='oof_pred_bag'`) and for this case `X_test` is definitely required.
